webapp/src/auth.rs

```diff
@@ -0,0 +1,128 @@
+use axum::async_trait;
+use axum::routing::get;
+use axum::routing::post;
+use axum::Router;
+use axum_login::AuthUser;
+use axum_login::AuthnBackend;
+use axum_login::UserId;
+use bitcoin::hashes::hex::ToHex;
+use serde::Deserialize;
+use sha2::digest::FixedOutput;
+use sha2::Digest;
+use sha2::Sha256;
+use std::error::Error;
+use std::fmt::Display;
+use std::fmt::Formatter;
+
+#[derive(Clone)]
+pub struct Backend {
+    pub(crate) hashed_password: String,
+}
+
+#[derive(Clone, Debug)]
+pub struct User {
+    password: String,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct Credentials {
+    pub password: String,
+}
+
+#[derive(std::fmt::Debug)]
+pub struct BackendError(String);
+
+impl Display for BackendError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        self.0.to_string().fmt(f)
+    }
+}
+
+impl Error for BackendError {}
+
+#[async_trait]
+impl AuthnBackend for Backend {
+    type User = User;
+    type Credentials = Credentials;
+    type Error = BackendError;
+
+    async fn authenticate(
+        &self,
+        creds: Self::Credentials,
+    ) -> Result<Option<Self::User>, Self::Error> {
+        let mut hasher = Sha256::new();
+        hasher.update(creds.password.as_bytes());
+        let hashed_password = hasher.finalize_fixed().to_hex();
+
+        let user = match hashed_password == self.hashed_password {
+            true => Some(User {
+                password: self.hashed_password.clone(),
+            }),
+            false => None,
+        };
+
+        Ok(user)
+    }
+
+    async fn get_user(&self, _user_id: &UserId<Self>) -> Result<Option<Self::User>, Self::Error> {
+        Ok(Some(User {
+            password: self.hashed_password.clone(),
+        }))
+    }
+}
```

Comment:
Can we still use curl to test the endpoints?

Reply:
yes.